aloha/real_env.py

```diff
@@ -241,38 +270,52 @@ def reset(self, fake=False):
             observation=self.get_observation(),
         )
 
-    def step(self, action, base_action=None, get_base_vel=False, get_obs=True):
-        state_len = int(len(action) / 2)
-        left_action = action[:state_len]
-        right_action = action[state_len:]
-        self.follower_bot_left.arm.set_joint_positions(left_action[:6], blocking=False)
-        self.follower_bot_right.arm.set_joint_positions(right_action[:6], blocking=False)
-        self.set_gripper_pose(left_action[-1], right_action[-1])
+    def step(self, action, base_action=None, get_obs=True):
+
+        # Dynamically calculate per-bot state length
+        state_len = int(len(action) / len(self.follower_robots))
+        index = 0
+
+        # Iterate through each follower bot and set joint positions
+        for name, robot in self.follower_robots.items():
+            bot_action = action[index:index + state_len]
+            robot.arm.set_joint_positions(
+                bot_action[:6], blocking=False)  # Set arm positions
```

Comment:
Allows us to use arms of different size assuming that actions look like [joint_0, joint_1, ..., joint_n, gripper]

```suggestion
                bot_action[:-1], blocking=False)  # Set arm positions
```